include/zephyr/drivers/sensor.h

```diff
@@ -313,6 +320,27 @@ enum sensor_attribute {
 	/** Alert threshold or alert enable/disable */
 	SENSOR_ATTR_ALERT,
 
+	/**
+	 * To enroll fingerprint image as template
+	 */
+	SENSOR_ATTR_ADD,
+	/**
+	 * To search fingerprint template
+	 */
+	SENSOR_ATTR_FIND,
+	/**
+	 * To delete fingerprint template or to empty template database
+	 */
+	SENSOR_ATTR_DEL,
+
+	/**
+	 * To get free Fingerprint template index for enrolling
+	 */
+	SENSOR_ATTR_FREE_IDX,
+	/**
+	 * To empty Fingerprint library
+	 */
+	SENSOR_ATTR_EMPTY,
```

Comment:
how do I know these are attributes for a fingerprint "SENSOR_ATTR_ADD"? I'd suggest using custom driver attributes here.

Reply:
understood and added generic attributes SENSOR_ATTR_RECORD_ which will be common to any sensor which is having internal storage.